### PR TITLE
[FLINK-35245][cdc-connector][tidb] Add metrics for flink-connector-tidb-cdc

### PR DESCRIPTION
**Pull request by xieyi888**
https://issues.apache.org/jira/browse/FLINK-35245

add metrcis: currentFetchEventTimeLag, currentEmitEventTimeLag, sourceIdleTime for TiKVRichParallelSourceFunction

### CODE REVIEW

**Review by yuxiqian — APPROVED** [review]
Thanks for @xieyi888's quick response! Pulled, ran E2e tests, and the metrics look good 😊
Metrics running result

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-tidb-cdc/src/test/java/org/apache/flink/cdc/connectors/tidb/metrics/TiDBSourceMetricsTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Unit test for {@link TiDBSourceMetrics}. */
+public class TiDBSourceMetricsTest {
+    private static final String FETCH_EVENTTIME_LAG = "currentFetchEventTimeLag";
```
[the remark this answers is not in the document]
Comment (xieyi888) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-tidb-cdc/src/test/java/org/apache/flink/cdc/connectors/tidb/metrics/TiDBSourceMetricsTest.java]: > Maybe can use MetricNames.CURRENT_FETCH_EVENT_TIME_LAG Replace it?

Thanks @czy006 , I had use MetricNames.CURRENT_EMIT_EVENT_TIME_LAG,
MetricNames.CURRENT_FETCH_EVENT_TIME_LAG,
MetricNames.SOURCE_IDLE_TIME
instead

**Comment by xieyi888** [comment]
@yuxiqian @czy006  I had made changes, Please review

**Review by czy006 — APPROVED** [review]
LGTM

**Review by Jiabao-Sun — APPROVED** [review]
+1

**Pull request merged**